src/server/env.ts

```diff
@@ -0,0 +1,24 @@
+import { config } from 'dotenv'
+import makeParser from '../lib/makeParser'
+
+config()
+
+interface Env {
+  LOG_LEVEL: string
+  LOG_PRETTY: boolean
+  NODE_ENV: 'development' | 'production' | 'test'
+  PORT: number
+}
+
+const parse = makeParser<Env>({
+  type: 'object',
+  properties: {
+    LOG_LEVEL: { type: 'string', default: 'info' },
+    LOG_PRETTY: { type: 'boolean', default: false },
+    NODE_ENV: { type: 'string', default: 'development' },
+    PORT: { type: 'number', default: 3000 },
+  },
+  required: [],
+})
+
+export default parse((process.env as unknown) as Env)
```

Comment:
ok, I'm just wondering.. `process.env` interface is actually `Record<string, string>`. Would this affect at all this implementation? Could this make the use of this a bit confusing?

Also, we might be able to change the types in the `makeParser` funct, to make it clearer.